src/services/MusicService.js

```diff
@@ -77,4 +77,29 @@ export class MusicService {
       )
     );
   }
+
+  static getSongRating(userId) {
+    return database
+      .ref(`users/${userId}/rating`)
+      .once("value")
+      .then(rating => rating.val());
+  }
+
+  static setNewRating(userId, songId, ratingValue) {
+    database
+      .ref(`users/${userId}`)
+      .once("value")
+      .then(user => user.val().rating)
+      .then(rating => {
+        if (!rating.includes(songId)) {
+          database.ref(`users/${userId}/rating/${rating.length}`).set(songId);
```

Comment:
as far as I understand - database.ref().set() is async operation - please synchronise these changes. Moreover - please `return` correct promises from if and else